src/group_theory/free_abelian_group.lean

```diff
@@ -345,7 +398,68 @@ instance [comm_monoid α] : comm_ring (free_abelian_group α) :=
       { intros y1 y2 ih1 ih2, rw [mul_add, add_mul, ih1, ih2] } },
     { intros s ih, rw [neg_mul_eq_neg_mul_symm, ih, neg_mul_eq_mul_neg] },
     { intros x1 x2 ih1 ih2, rw [add_mul, mul_add, ih1, ih2] }
-  end
+  end,
   .. free_abelian_group.ring α }
 
+/-- The free abelian group on the empty type is the trivial group. -/
+def pempty_equiv : free_abelian_group pempty ≃+ unit :=
+{ to_fun := lift (λ _, ()),
+  inv_fun := λ _, 0,
+  left_inv := λ z, free_abelian_group.induction_on z
+    ( rfl )
+    ( λ x, pempty.elim x)
+    ( λ x, pempty.elim x)
+    ( λ x y hx hy, by {simp only * at *, simp * at *}),
+  right_inv := λ z, unit.ext,
+  map_add' := λ x y, unit.ext }
+
+/-- The free abelian group on `punit` is isomorphic to `ℤ`. -/
+def punit_equiv : free_abelian_group punit ≃+ ℤ :=
+{ to_fun := free_abelian_group.lift (λ _, (1 : ℤ)),
```

Comment:
This generalizes to `unique T` instead of `punit`, which would cover `fin 1` too.

Reply:
Thanks.